x-pack/apm-server/sampling/processor_test.go

```diff
@@ -905,3 +905,11 @@ func waitFileModified(tb testing.TB, filename string, after time.Time) ([]byte,
 		}
 	}
 }
+
+func cleanupProcessor(t *testing.T, processor *sampling.Processor) {
```

Comment:
WDYT about a new function which both creates and starts the processor, and adds a cleanup to stop it with immediate cancellation?

e.g.

```go
func newProcessor(t testing.TB, cfg sampling.Config) *sampling.Processor
```